Dashboard: Unified Menu

kean opened the pull request:
Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/21492

I kept the QuickActions cell but changed the layout to vertical and updated the cell design. The rest of the code is largely the same. It was also able to remove four iOS 15 deprecations, which is nice. And the dashboard is now vastly simplified.

I also added a way to remove Quick Start tour for a current blog.

### Not in the Scope

- Update the blog details page (more menu)
- Adding details labels to the new menus items (it will require more work and will be done separately)

<img width="453" alt="Screenshot 2023-09-12 at 10 45 19 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/d65bc89e-ca11-44b9-84a4-19c6c82b4c47">

## To test:

- Verify that the cell is highlighted as you select it and is deselected when you return to the dashboard: [recording](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/a5e31da1-9a97-4eea-9e74-928c20a9a20a)

### Site with no Jetpack

- Add a self-hosted site without Jetpack and verify that the app shows a "more" menu instead of the dashboard
- Select a WP.COM site from a site picker and verify that it switched to dashboard

### Quick Start

- Verify that the app leads you through the entire flow
- Verify that the new unified menu has spotlights when needed: [recording](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/ba3c3b1f-01cc-48d5-bf21-57363294f29c)
- Verify that tours that require you to open blog details prompt you to tap "More": [recoding](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/55706b52-71bc-45ef-a4a6-db6492a918b3)


### WordPress
- Verify that WordPress is unaffected

## Regression Notes
1. Potential unintended areas of impact: Dashboard and Menu
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual tests
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [x] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Remove segmented control from dashboard

kean committed:
Add initial DashboardMenu implementation

kean committed:
Intial dashboard menu implementation

kean committed:
Remove MePresenter from BlogDetailsVC

kean committed:
Implement new dashboard menu

kean committed:
Implement spotlight

kean committed:
Hide quick actions on iPad

kean committed:
Fix create sheet

kean committed:
Add a way to remove current Quick Start tour

kean committed:
Fix an issue with More tours no longer working

kean committed:
Fix remaining Quick Start flows

kean commented on WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Quick Actions/DashboardQuickActionsViewModel.swift:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    func viewWillDisappear() {
+
```
I'll need it when I add the details label.

kean commented on WordPress/Classes/ViewRelated/Blog/QuickStartToursCollection.swift:
```diff
@@ -59,8 +59,8 @@ struct QuickStartGrowToursCollection: QuickStartToursCollection {
         self.tours = [
-            QuickStartShareTour(),
             QuickStartPublishTour(),
+            QuickStartShareTour(),
```
I moved these because the previous order didn't flow well anymore.

momo-ozawa replied:
Sounds good! Can you also update the Share tour step 1 description target from "Sharing" to "Social"? We should have updated this when we first changed the copy, but I only noticed it now 🙇‍♀️ 

```
let step1DescriptionTarget = NSLocalizedString("Social", comment: "The menu item to select during a guided tour.")
```

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/a19ae829-98e2-4f80-9ed0-7c1f9c82e58a" width=200>

momo-ozawa replied:
I also just noticed that we should also probably update the Quick Start icons to use the new assets - but feel free to address this later

kean committed:
Fix dark mode

kean committed:
Update tests

kean committed:
Remove unused siteMenuSpotlight

kean committed:
Update site header design

kean committed:
Update the Pages icon

kean committed:
Fix an issue with incorrect loading indicator position

kean committed:
Fix dark mode support

kean committed:
Add comments menu item

kean committed:
Implement quick actions personalizations

jostnes committed:
split mySiteScreen to two screens

jostnes committed:
rename menu to more menu

kean committed:
Update copy

jostnes committed:
Merge pull request #21530 from wordpress-mobile/fix-ui-tests

momo-ozawa reviewed (commented):
Looking good, @kean!
 
### Quick start 
- The pages quick start tour doesn't scroll to the pages item

### More screen
Out of scope for this PR, but jotting down notes while its fresh in my mind

- The nav title for the More screen is "My Site". A bit confusing because the previous screen is also called "My Site"
- The nav bar looks off when you go to tap More then tap My Site. I think you might want to set `extendedLayoutIncludesOpaqueBars = true` in `BlogDetailsViewController`


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/62e595e7-330d-41b8-be70-b310c9f94b5b

kean committed:
Fix an issue with pop animation

kean committed:
Fix an issue with quick start not scrolling to top

kean commented:
> The pages quick start tour doesn't scroll to the pages item

Fixed

> Out of scope for this PR, but jotting down notes while its fresh in my mind

Fixed both of these two. Thank you for the tip!

The final version looks like this: nice and sharp:

<img width="453" alt="Screenshot 2023-09-08 at 10 17 24 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/77f47fd5-7a58-4491-9c42-3f5a087efd4c">

kean committed:
Tweak my site appearance

momo-ozawa commented on WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+QuickStart.swift:
```diff
@@ -13,15 +13,18 @@ extension MySiteViewController {
         }
 
+        switch element {
+        case .siteIcon, .siteTitle, .viewSite, .pages, .stats, .mediaScreen:
```
I believe .sharing should also be added here, since it's accessed via the new More item

kean replied:
Updated, thanks!

kean committed:
Enable scrolling for sharing

kean commented:
Note on merging: I'm planning to merge this when the rest of the PRs are also approved.

kean committed:
Make quick actions personalization site-specific

kean committed:
Reorder quick action and put stats on top

kean committed:
Show shortcut customization based on eligibility

kean committed:
Merge pull request #21523 from wordpress-mobile/task/unified-menu-site-header

Dashboard: update site header design

kean committed:
Refresh blog menu design

kean committed:
Add context menu to subtitle

kean committed:
Update copy

kean committed:
Fix an issue with personalization not working when switching sites

kean committed:
Merge pull request #21536 from wordpress-mobile/task/unified-menu-finishing-touches

Dashboard: Unified Menu finishing touches

kean committed:
Fix tests

kean committed:
Merge pull request #21526 from wordpress-mobile/task/unitifed-menu-personalization

Dashboard: Unified Menu Personalization

kean committed:
Merge pull request #21537 from wordpress-mobile/task/add-context-menu-url

Dashboard: add context menu to site link

momo-ozawa commented on WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+QuickStart.swift:
```diff
@@ -13,17 +13,18 @@ extension MySiteViewController {
 
-        siteMenuSpotlightIsShown = element == .siteMenu
+        switch element {
+        case .siteIcon, .siteTitle, .viewSite, .pages, .stats, .mediaScreen, .sharing:
```
Turns out we also need to add .siteMenu to get it to scroll for the social sharing tour
```suggestion
        case .siteIcon, .siteTitle, .viewSite, .pages, .stats, .mediaScreen, .sharing, .siteMenu:
```

momo-ozawa reviewed (approved):
LGTM w a few minor suggestions
